### PR TITLE
Add information about on demand repository cleaning

### PR DESCRIPTION
**Pull request by AkshayGadhaveRH**
Adding information about reclaiming space with on demand repo cleaning. Demoed in  Foreman community demo #105.

Cherry-pick into:

* [ ] Foreman 3.3
* [ ] Foreman 3.2
* [ ] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

### CODE REVIEW

File: guides/common/assembly_maintaining-server.adoc [hunk 1]
```diff
@@ -17,3 +17,7 @@ endif::[]
 ifndef::foreman-deb[]
 include::modules/proc_reclaiming-postgresql-space.adoc[leveloffset=+1]
 endif::[]
+
+ifdef::katello,,satellite[]
```
Comment (melcorr) [inline guides/common/assembly_maintaining-server.adoc:21]: What happened here?

File: guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
+[id="reclaiming-space-from-on-demand-repositories_{context}"]
+= Reclaiming Space From On Demand Repositories
+
+When the download policy is set to on demand, the actual packages are downloaded only when the clients request them.
```
Comment (melcorr) [inline guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc:4]: Is actual needed here ?
```suggestion
When the download policy is set to on demand, the packages are downloaded only when the clients request them.
```

File: guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+These packages can be cleaned up to reclaim space.
```
Comment (melcorr) [inline guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc:5]: Rephrase this to avoid the passive voice.

**Comment by AkshayGadhaveRH** [comment]
Hey @melcorr Ian Ballou (the SME who demoed this) has reviewed and approved the content. I'll get peer approval for this too.

File: guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
+[id="reclaiming-space-from-on-demand-repositories_{context}"]
+= Reclaiming Space From On Demand Repositories
+
+When the download policy is set to on demand, the {Project} downloads packages only when the clients request them.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc:4]:
```suggestion
If you set the download policy to _on demand_, {Project} downloads packages only when clients request them.
```

File: guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+You can clean up these packages to reclaim space.
+
+.For a single repository
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc:8]:
```suggestion
```

You don't need an empty live below headings starting with a dot. Please fix all instances.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Couple of minor suggestions.

**Comment by maximiliankolb** [comment]
Hi @AkshayGadhaveRH You have resolved some threads without making a change as suggested. If you  don't want to use a suggestions (which is fine!), please state why.

I will happily review this again after you have addressed and closed all open threads in this PR. Please also rebase to master and have a look at your commit message(s). Having two or more commits on a PR is perfectly fine if there are multiple changes that are not directly connected to each other. Otherwise, please squash your commits. See also https://cbea.ms/git-commit/#seven-rules

Thank you!

**Comment by AkshayGadhaveRH** [comment]
Hey @maximiliankolb my bad, let me check once again. I think because I committed the earlier suggestions remotely, and the later commit was pushed from my local branch, I suspect the earlier commits got overwritten with the files that did not have these changes. Let me go over all the suggestions once again. Sorry for the inconvenience.

**Comment by maximiliankolb** [comment]
No worries, ping me when you're ready.

**Comment by AkshayGadhaveRH** [comment]
Hi @maximiliankolb the suggestions did get overwritten because I committed from the local branch where I had not applied the earlier suggestions. I have done so now and pushed a new commit. Once again, sorry for the inconvenience, and thanks for your patience.

File: guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc [hunk 6]
```diff
@@ -0,0 +1,22 @@
+[id="reclaiming-space-from-on-demand-repositories_{context}"]
+= Reclaiming Space From On Demand Repositories
+
+If you set  the _download policy_ to on demand, {Project} downloads packages only when the clients request them.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_reclaiming-space-from-on-demand-repositories.adoc:4]:
```suggestion
If you set the _download policy_ to on demand, {Project} downloads packages only when the clients request them.
```

**Review by maximiliankolb — COMMENTED** [review]
I closed all open threads for you. One very minor change, then we're good to go and I will merge.
